src/Emoji.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "Singleton.hpp"
+#include "types.hpp"
+
+#include <string>
+#include <atomic>
+#include <vector>
+#include <json.hpp>
+
+using json = nlohmann::json;
+
+class Emoji
+{
+public:
+	Emoji(Snowflake_t const & id, std::string const & name);
+	~Emoji() = default;
+
+	inline Snowflake_t const& GetSnowflake()
+	{
+		return _id;
+	}
+	inline std::string const& GetName()
+	{
+		return _name;
+	}
+private:
+	Snowflake_t _id;
+	std::string _name;
+
+};
+
+class EmojiManager : public Singleton<EmojiManager>
+{
+	friend class Singleton<EmojiManager>;
+private:
+	EmojiManager() = default;
+	~EmojiManager() = default;
+
+private:
+	const unsigned int m_InitValue = 1;
+	std::atomic<unsigned int> m_Initialized{ 0 };
```

Comment:
You can safely remove those two variables, they're used in other resource managers for initial cache filling on plugin startup.